core/sail/api/src/main/java/org/eclipse/rdf4j/sail/helpers/AbstractSailConnection.java

```diff
@@ -459,6 +459,10 @@ public final void rollback()
 						txnPrepared = false;
 					}
 				}
+				else {
+					logger.warn("Cannot rollback transaction on connection because transaction is not active",
+							new Throwable());
```

Comment:
Looks like that may need a bit of revision (for one thing the system property should be renamed), as well as documentation.